php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+import static org.sonar.php.api.PHPPunctuator.NOTEQUALBIS;
+import static org.sonar.php.api.PHPPunctuator.PLUS;
+import static org.sonar.php.api.PHPPunctuator.QUERY;
+import static org.sonar.php.api.PHPPunctuator.RBRACKET;
+import static org.sonar.php.api.PHPPunctuator.RCURLYBRACE;
+import static org.sonar.php.api.PHPPunctuator.RPARENTHESIS;
+import static org.sonar.php.api.PHPPunctuator.SL;
+import static org.sonar.php.api.PHPPunctuator.SR;
+import static org.sonar.php.api.PHPPunctuator.STAR;
+
+public class NewPHPGrammar {
+
+  private final GrammarBuilder<InternalSyntaxToken> b;
+  private final TreeFactory f;
+
+  public NewPHPGrammar(GrammarBuilder<InternalSyntaxToken> b, TreeFactory f) {
+    this.b = b;
+    this.f = f;
+  }
+
+  public CompilationUnitTree COMPILATION_UNIT() {
+    return b.<CompilationUnitTree>nonterminal(PHPLexicalGrammar.COMPILATION_UNIT).is(
+      f.compilationUnit(b.optional(SCRIPT()), b.optional(b.token(PHPLexicalGrammar.SPACING)), b.token(PHPLexicalGrammar.EOF)));
+  }
+
+  public ScriptTree SCRIPT() {
+    return b.<ScriptTree>nonterminal(PHPLexicalGrammar.SCRIPT).is(
+      f.script(b.token(PHPLexicalGrammar.FILE_OPENING_TAG), b.zeroOrMore(TOP_STATEMENT())));
+  }
+
+  /**
+   * [ START ] Declaration
+   */
+
+  public VariableDeclarationTree MEMBER_CONST_DECLARATION() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.MEMBER_CONST_DECLARATION).is(
+      f.memberConstDeclaration(
+        b.token(PHPLexicalGrammar.IDENTIFIER),
+        b.optional(f.newTuple18(b.token(EQU), STATIC_SCALAR()))));
+  }
+
+  public VariableDeclarationTree CONST_VAR() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.CONSTANT_VAR).is(
+      f.constDeclaration(
+        b.token(PHPLexicalGrammar.IDENTIFIER),
+        b.token(EQU),
+        STATIC_SCALAR()));
+  }
+
+  public VariableDeclarationTree VARIABLE_DECLARATION() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.VARIABLE_DECLARATION).is(
+      f.variableDeclaration(
+        b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+        b.optional(f.newTuple96(b.token(EQU), STATIC_SCALAR()))));
+  }
+
+  public NamespaceNameTree NAMESPACE_NAME() {
+    return b.<NamespaceNameTree>nonterminal(PHPLexicalGrammar.NAMESPACE_NAME).is(
+      b.firstOf(
+        f.namespaceName(
+          b.optional(b.token(PHPPunctuator.NS_SEPARATOR)),
+          b.zeroOrMore(f.newTuple4(
+            b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
+            b.token(PHPPunctuator.NS_SEPARATOR)
+          )),
+          b.token(PHPLexicalGrammar.IDENTIFIER)),
+        f.namespaceName(
+          b.token(PHPKeyword.NAMESPACE),
+          b.token(PHPPunctuator.NS_SEPARATOR),
+          b.zeroOrMore(f.newTuple6(
+            b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
+            b.token(PHPPunctuator.NS_SEPARATOR)
+          )),
+          b.token(PHPLexicalGrammar.IDENTIFIER))));
+  }
+
+  public UseClauseTree USE_CLAUSE() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)